spawn/__init__.py

```diff
@@ -26,9 +26,14 @@
 __copyright__ = 'Copyright (C) 2018, Simmovation Ltd.'
 __author__ = 'Simmovation Ltd.'
 
+from .plugins import json_input_file as json_plugin
+from .plugins import PluginLoader
 from ._version import get_versions
 __version__ = get_versions()['version']
 del get_versions
 
 # Default `run` and `inspect` should be local
 from .interface import run_local as run, inspect_local as inspect, write_inspection
+
+# Load built-in plugins
+PluginLoader.pre_load_plugin('json', json_plugin)
```

Comment:
Do we want to couple the default implementation of the json plugin to an exe runner? Or perhaps we want another json plugin that just writes the files? Maybe we get around this by making this name more specific e.g. `json_input_file`

Reply:
I made changes in `SimulationTask` so that if the `_exe_path` is not specified (empty) then it won't actually run any processes and therefore just write the file. Therefore the default behaviour of the `json` plugin if no `exe_path` config parameter is specified is to just write the files. So I think the `json` plugin serves both purposes

Reply:
Although maybe I've confused things by calling the plugin file `json_input_file.py` but then just calling the plugin `json`. I called it `json_input_file` because it's the input files that are JSON (not the simulations per se), but then simplified the plugin name.

I guess given that the exe is always arbitrary (user defined) when we talk about spawn plugins it's implied that we're always talking about the input file generation or simulation parameterisation. For example if we talk about NREL plugin we're talking about the control of NREL input files - they could make the FAST exe anything they like. What do you think?

Reply:
I guess what we're providing is sort of a "short cut" to run an arbitrary exe with an arbitrary input file. I don't really see a problem with that. If the user wants to do anything more complicated with input files then we won't be able to guess what that is and shouldn't try. We might be able to e.g. template exe commands, but that's a story for another PR...